CHANGELOG.md

```diff
@@ -1,25 +1,223 @@
 ## vNext (TBD)
+NOTE: This combines all changelog entries for prereleases of v12.0.0.
+
+### Breaking changes
+Although this is a complete rewrite of our SDK, we've strived to keep breakages to a minimum and expect our users to upgrade from v11 without any significant changes to their code-base.
+
+* The entire BSON package used to be re-exported as `Realm.BSON`, to simplify the new SDK we want to export only the BSON types that our SDK database component supports (ObjectId, Decimal128 and UUID). See [#4934](https://github.com/realm/realm-js/issues/4934).
+* We're now reusing code to perform assertions and although this is strictly not a breaking change, since we haven't historically documented error messages, you should probably revisit any code in your app which relies on matching on specific error messages.
+* `Results`, `List` and `Set` used to inherit directly from `Collection` but now inherits from an abstract `OrderedCollection`, which extends `Collection`.
+* In order to better guide users toward correct usage and understanding of the Realm property types, users must now be explicit about the property type when declaring object schemas. Additionally, mixing shorthand (string) and object representation for the property type is no longer permitted. (See the `PropertySchema` and `PropertySchemaShorthand` types.)
+
+```javascript
+// Example object schema
+const TaskSchema = {
+  name: "Task",
+  properties: {
+    description: /* property schema (shorthand or object form) */,
+  },
+};
+
+// Explicitness
+"[]"          // Bad (previously parsed as implicit "mixed")
+"mixed[]"     // Good
+
+{ type: "list" }                              // Bad
+{ type: "list", objectType: "mixed" }         // Good
+
+// Mixing shorthand and object form
+{ type: "int[]" }                             // Bad
+"int[]"                                       // Good
+{ type: "list", objectType: "int" }           // Good
+
+{ type: "int?" }                              // Bad
+"int?"                                        // Good
+{ type: "int", optional: true }               // Good
+
+// Specifying object types
+{ type: "SomeType" }                          // Bad
+"SomeType"                                    // Good
+{ type: "object", objectType: "SomeType" }    // Good
+
+{ type: "object[]", objectType: "SomeType" }  // Bad
+"SomeType[]"                                  // Good
+{ type: "list", objectType: "SomeType" }      // Good
+
+{ type: "linkingObjects", objectType: "SomeType", property: "someProperty" } // Good
+```
+
+* To prevent modifying end-users' class-based model classes, we’re now wrapping class-based models in our own model. Objects will still pass `instanceof SomeClass` checks, however, code which is directly using prototype or constructor comparisons will fail:
```

Comment:
```suggestion
* To prevent modifying end-users' class-based model classes, we’re now creating and injecting a class in front of the class provided by the user. Objects will still pass `instanceof SomeClass` checks, however, code which is directly using prototype or constructor comparisons will fail:
```